lisa/features/security_profile.py

```diff
@@ -32,7 +33,11 @@ class SecurityProfile(Feature):
     @classmethod
     def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
         settings = cast(SecurityProfileSettings, kwargs.get("settings"))
-        if settings.secure_boot_enabled:
+        runbook: Any = cast(Any, kwargs.get("arm_parameters")).nodes[0]
```

Comment:
It's the arm_parameters, not runbook. The "settings" is the right place to check the field, although they are the same source.

Reply:
Is there a better location for setting SecurityProfileSettings based on "arm_parameters"?

Is your suggestion to just rename the variable `runbook` to `arm_parameters`?

Reply:
I mean it should use the setting to check the field. It doesn't need the arm_template.

Reply:
Is it changed to use settings? Besides that, it needs to update runbook of each node to use the deployed type of CVM.